HW_6/task_2.js

```diff
@@ -0,0 +1,271 @@
+          name: "Отдел разработки",
+          employees_count: 50,
+        },
+        {
+          id: 7,
+          name: "Отдел маркетинга",
+          employees_count: 20,
+        },
+        {
+          id: 8,
+          name: "Отдел охраны труда",
+          employees_count: 5,
+        },
+      ]
+    },
+    {
+      id: 9,
+      name: "Предприятие 3",
+      departments: [
+        {
+          id: 10,
+          name: "Отдел аналитики",
+          employees_count: 0,
+        },
+      ]
+    }
+  ];
+
+  function task1(obj) {
+    obj.forEach(firm => {  
+        const totalEmployees = sumEmployees(firm.departments);              
+        console.log(`${firm.name} (${totalEmployees})`);
+        firm.departments.forEach(department => {
+            const employeesWord = getEmployeesWord(department.employees_count);
+            console.log(`  ${department.name} (${department.employees_count} ${employeesWord})`);
+        });
+    });
+};
+
+
+function sumEmployees(departments) {
+    return departments.reduce((sum, department) => sum + department.employees_count, 0);
+};
+
+
+function getEmployeesWord(count) {
+    if (count % 10 === 1 && count % 100 !== 11) {
+        return 'сотрудник';
+    } else if ([2, 3, 4].includes(count % 10) && ![12, 13, 14].includes(count % 100)) {
+        return 'сотрудника';
+    } else {
+        return 'сотрудников';
+    }
+};
+
+task1(enterprises);
+
+
+
+
+// 2. Написать функцию, которая будет принимать 1 аргумент (id отдела или название отдела и возвращать название предприятия, к которому относится).
+
+// Пример:
+// getEnterpriseName(4) // Предприятие 1
+// getEnterpriseName("Отдел маркетинга") // Предприятие 2
+
+function getEnterpriseName(idOrName) {
+    const foundEnterpeise = getEnterpriseByDepartment(idOrName);
+    return foundEnterpeise.name;
+  }
+  
+  function getEnterpriseByDepartment(idOrName) {
+    const foundEnterpeise = enterprises.find((enterprise) => {
+      const foundDeparment = enterprise.departments.find((department) => department.name === idOrName || department.id === idOrName);
+      return foundDeparment;
+    });
+    if (!foundEnterpeise) throw new Error("No such Enteprise");
+    return foundEnterpeise;
+  };
+
+function getEnterpriseByIdOrName(idOrName){
+    const foundEnterpeise = enterprises.find((enterprise)=> enterprise.id === idOrName || enterprise.name === idOrName);
+    return foundEnterpeise;
+};
+
+console.log(getEnterpriseName(2));
+console.log(getEnterpriseName("Отдел маркетинга"));
+
+// 3. Написать функцию, которая будет добавлять предприятие. В качестве аргумента принимает название предприятия
+
+// Пример:
+// addEnterprise("Название нового предприятия")
+
+function addEnterprise(nameEnterprise) {    
+    const maxId = sumIds(enterprises)   
+    let newObj = {name: nameEnterprise, id: maxId + 1, departments: []};    
+    enterprises.push(newObj);
+    
+    return getEnterpriseByIdOrName(nameEnterprise);
+  };
+
+function sumIds() {

```

Comment:
`getNewID` || `calculateNewID` || `generateNewID` || `retrieveMaxID` || `findMaxID`
```suggestion
function sumIds() {
```